test/OrchardCore.Benchmarks/FluidShapeRenderBenchmark.cs

```diff
@@ -0,0 +1,161 @@
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using Fluid;
+using Fluid.Values;
+using Microsoft.AspNetCore.Html;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.DisplayManagement;
+using OrchardCore.DisplayManagement.Implementation;
+using OrchardCore.DisplayManagement.Liquid;
+using OrchardCore.DisplayManagement.Liquid.Filters;
+using OrchardCore.Liquid;
+
+namespace OrchardCore.Benchmark
+{
+    [MemoryDiagnoser]
+    public class FluidShapeRenderBenchmark
+    {
+        private static readonly FilterArguments _filterArguments = new FilterArguments().Add("utc", new DateTimeValue(DateTime.UtcNow)).Add("format", StringValue.Create("MMMM dd, yyyy"));
+        private static readonly FluidValue _input = ObjectValue.Create(HtmlString.Empty, new TemplateOptions());
+        private static readonly LiquidFilterResolver<ShapeRenderFilter> _liquidFilterResolver;
+        private static readonly IServiceProvider _serviceProvider;
+
+        static FluidShapeRenderBenchmark()
+        {
+            var htmlDisplay = new DefaultHtmlDisplay(null, null, null, null, null, null);
+
+            _serviceProvider = new ServiceCollection()
+                .AddScoped<IDisplayHelper>(sp => new DisplayHelper(htmlDisplay, null, null))
+                .AddTransient(typeof(ShapeRenderFilter))
+                .BuildServiceProvider();
+
+            _liquidFilterResolver = new LiquidFilterResolver<ShapeRenderFilter>();
+        }
+
+        // 21st Feb 2021
+
+        // BenchmarkDotNet=v0.12.1, OS=macOS Catalina 10.15.7 (19H15) [Darwin 19.6.0]
+        // Intel Core i7-6700K CPU 4.00GHz (Skylake), 1 CPU, 8 logical and 4 physical cores
+        // .NET Core SDK=5.0.103
+        //   [Host]     : .NET Core 3.1.8 (CoreCLR 4.700.20.41105, CoreFX 4.700.20.41903), X64 RyuJIT
+        //   DefaultJob : .NET Core 3.1.8 (CoreCLR 4.700.20.41105, CoreFX 4.700.20.41903), X64 RyuJIT
+
+
+        // |                       Method |     Mean |     Error |    StdDev | Ratio | RatioSD |  Gen 0 | Gen 1 | Gen 2 | Allocated |
+        // |----------------------------- |---------:|----------:|----------:|------:|--------:|-------:|------:|------:|----------:|
+        // |   OriginalShapeRenderDynamic | 3.798 us | 0.0663 us | 0.1108 us |  1.00 |    0.00 | 2.2888 |     - |     - |   9.37 KB |
+        // | ShapeRenderWithAmbientValues | 3.156 us | 0.0278 us | 0.0232 us |  0.82 |    0.03 | 2.2240 |     - |     - |   9.09 KB |
+        // |            ShapeRenderStatic | 3.180 us | 0.0253 us | 0.0224 us |  0.83 |    0.03 | 2.1935 |     - |     - |   8.96 KB |
+        // |      ShapeRenderWithResolver | 3.164 us | 0.0206 us | 0.0161 us |  0.83 |    0.03 | 2.1973 |     - |     - |   8.98 KB |
```

Comment:
so resolving adds a tiny bit of allocation (for the instance), but constructor is marginally faster than `serviceprovider.GetService()`

Changed these benchmarks from what you saw yesterday to simulate creating a template context as well. (as we would have done each liquid render).